v4/export/sql.go

```diff
@@ -514,8 +514,20 @@ func isUnknownSystemVariableErr(err error) bool {
 func resetDBWithSessionParams(tctx *tcontext.Context, db *sql.DB, dsn string, params map[string]interface{}) (*sql.DB, error) {
 	support := make(map[string]interface{})
 	for k, v := range params {
+		var pv interface{}
+		if str, ok := v.(string); ok {
+			if pvi, err := strconv.ParseInt(str, 10, 64); err == nil {
+				pv = pvi
+			} else if pvf, err := strconv.ParseFloat(str, 64); err == nil {
+				pv = pvf
+			} else {
+				pv = str
+			}
+		} else {
+			pv = v
+		}
 		s := fmt.Sprintf("SET SESSION %s = ?", k)
```

Comment:
Here using `?` will let sql driver to check the type of `pv`, to decide if `pv` need quotes.

Instead, I prefer always use the input value as-is, that is to say we always use `%s` and printf to construct a SQL. So user should quote the value if it's a string type variable in SQL.

https://github.com/go-sql-driver/mysql/blob/bcc459a906419e2890a50fc2c99ea6dd927a88f2/connection.go#L68-L78

Reply:
we might append `--param` to dsn 😂 otherwise we can't control https://github.com/go-sql-driver/mysql#parameters and have to write a doc about dealing with `,` in parameter KVs.

Reply:
> Here using `?` will let sql driver to check the type of `pv`, to decide if `pv` need quotes.
> 
> Instead, I prefer always use the input value as-is, that is to say we always use `%s` and printf to construct a SQL. So user should quote the value if it's a string type variable in SQL.
> 
> https://github.com/go-sql-driver/mysql/blob/bcc459a906419e2890a50fc2c99ea6dd927a88f2/connection.go#L68-L78
> 
> (I deleted previous misleading comment)

`conf.SessionParams` is a `map[string]interface{}`. I think using an original value to construct SQL is in-proper for being used as a library.

Reply:
what's the detailed in-proper case?

if user input `key=val`, successful parsing it as a integer does not mean that's an interger MySQL variable, at that time it will raise an error.

Reply:
I think that `SET X = 123` and `SET X = '123'` are equivalent

Reply:
@lance6716 @kennytm 
Fixed in https://github.com/pingcap/dumpling/pull/271/commits/1b33d20633bb27be2dfb1df4b069122b333a3cd5, and add an integration test in https://github.com/pingcap/dumpling/pull/271/commits/278a1289ac7825a963d2f0b86a8d857b9d40f889. PTAL again

Reply:
sorry I didn't reply to above comment in time 😂 I didn't found a case for system variables that `SET SESSION X = 123` will cause error if X is a string type variable. Will check it now.